src/perl/vdotest/VDOTest/Vdorecover.pm

```diff
@@ -213,11 +214,13 @@ sub testFullFilesystem {
                   );
   $dir2->pollUntilDone();
 
-  $self->syncDeviceIgnoringErrors();
-
   # The error syncing matches either No space left on device or Input/output
   # error, but distinguishing them is hard.
+  $self->syncDeviceIgnoringErrors();
+  $fs->unmount();
+  $device->runOnHost("sudo dmesg -c > /dev/null");
```

Comment:
I think it would be useful to run the `sudo dmesg -c >/dev/null` *before* we fill the device.  And then at this spot, drop the `> /dev/null` piece.  This allows us to end up seeing the error messages generated by the filesystem from running out of space.

Reply:
Resolved in update to PR, although I do not see any output in the logs from the dmesg call after the writes.